### PR TITLE
[SDL2] Updating port to latest 2.28.2

### PR DESCRIPTION
**Pull request by Honeybunch**
- [X] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [X] SHA512s are updated for each updated download
- [X] The "supports" clause reflects platforms that may be fixed by this new version
- [X] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [X] Any patches that are no longer applied are deleted from the port's directory.
- [X] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [X] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by Cheney-W** [comment]
```
CMake Error at /mnt/vcpkg-ci/installed/x64-linux/share/sdl2/SDL2staticTargets.cmake:60 (set_target_properties):
  The link interface of target "SDL2::SDL2-static" contains:
    ALSA::ALSA
  but the target was not found.
```
Many downstream ports have failed on x64-linux due to the above error. Could you please fix this issue? The error log is here:
https://dev.azure.com/vcpkg/public/_build/results?buildId=93150&view=artifacts&pathAsName=false&type=publishedArtifacts

**Comment by Honeybunch** [comment]
Attempting the fix from the other PR on an older version to see what would happen and that looks like a bad idea. Going to have to give that another go

**Comment by Honeybunch** [comment]
Looks like that didn't work. Still going to wait for CI to fail so I can get the exact logs and make sure it's the same issue. I have a different attempt to find the ALSA package ready to go if it is. I wasn't able to repro this on my WSL2 setup so I hope y'all don't mind me hammering the CI a bit here

**Comment by Honeybunch** [comment]
Well this hasn't been as easy as I had hoped it would. Not having a local repro means I'm kind of stabbing at things here. I'm open to any ideas if my latest change still doesn't work :)

**Comment by Honeybunch** [comment]
@Cheney-W 
Any chance the latest x64_android and x64_linux jobs could be kicked?

**Comment by Honeybunch** [comment]
I did a diff against the previous working version and this latest version to see what is different with ALSA and it looks like quite a bit on the CMake side. I wonder if this ever worked before since there doesn't seem to have been any sort of `find_package` call for ALSA in the older version that builds. This also seems to be the only package that's found by `MODULE`. Really confused why marking ALSA as a dependency in the vcpkg manifest didn't resolve this

**Comment by dg0yt** [comment]
I think that [4722d43](https://github.com/microsoft/vcpkg/pull/33283/commits/4722d43b0e9060257ac3bc2318dd50dd1776b3ce) was the right direction for the initial problem, except that it didn't carry a guard to limit it to builds which actually require `ALSA`, i.e. something like

~~~cmake
if("@ALSA_FOUND@")
    find_dependency(ALSA)
endif()
~~~

File: ports/sdl2/portfile.cmake [hunk 1]
```diff
@@ -16,6 +17,7 @@ vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
     FEATURES
         vulkan   SDL_VULKAN
         x11      SDL_X11
+        alsa     SDL_ALSA
```
Comment (Cheney-W) [inline ports/sdl2/portfile.cmake:20]: sdl2 has no feature named alsa.

**Review by Cheney-W — CHANGES_REQUESTED** [review]

File: ports/sdl2/portfile.cmake (hunk 1 above)
Comment (Honeybunch) [reply ports/sdl2/portfile.cmake:20]: Whack, I removed that and I don't have it in my local changes. I wonder why it's showing up here?

Comment (Cheney-W) [reply ports/sdl2/portfile.cmake:20]: You might have missed it when committing locally. 
You can git pull the content of the branch again and then make modifications.

Comment (Honeybunch) [reply ports/sdl2/portfile.cmake]: No I'm just dumb and read the wrong file :)

**Comment by Honeybunch** [comment]
Huh I don't know why it says that there is still a change requested since I think I resolved it?

**Review by Cheney-W — APPROVED** [review]

**Comment by lsl330** [comment]
SDL2.28.3release！

**Pull request merged**